src/meta/meta_bulk_load_service.cpp

```diff
@@ -66,25 +66,20 @@ void bulk_load_service::on_start_bulk_load(start_bulk_load_rpc rpc)
     auto &response = rpc.response();
     response.err = ERR_OK;
 
-    std::shared_ptr<app_state> app;
-    {
-        zauto_read_lock l(app_lock());
-
-        app = _state->get_app(request.app_name);
-        if (app == nullptr || app->status != app_status::AS_AVAILABLE) {
-            derror_f("app({}) is not existed or not available", request.app_name);
-            response.err = app == nullptr ? ERR_APP_NOT_EXIST : ERR_APP_DROPPED;
-            response.hint_msg = fmt::format(
-                "app {}", response.err == ERR_APP_NOT_EXIST ? "not existed" : "dropped");
-            return;
-        }
-
-        if (app->is_bulk_loading) {
-            derror_f("app({}) is already executing bulk load, please wait", app->app_name);
-            response.err = ERR_BUSY;
-            response.hint_msg = "app is already executing bulk load";
-            return;
-        }
+    std::shared_ptr<app_state> app = get_app(request.app_name);
+    if (app == nullptr || app->status != app_status::AS_AVAILABLE) {
+        response.err = app == nullptr ? ERR_APP_NOT_EXIST : ERR_APP_DROPPED;
+        auto hint = fmt::format(
+            "app {} is ", response.err == ERR_APP_NOT_EXIST ? "not existed" : "not available");
+        response.hint_msg = hint;
+        derror_f("{}", hint);
+        return;
+    }
+    if (app->is_bulk_loading) {
+        derror_f("app({}) is already executing bulk load, please wait", app->app_name);
+        response.err = ERR_BUSY;
+        response.hint_msg = "app is already executing bulk load";
```

Comment:
like L72, define hint_msg and reuse it to derror_f(hint_msg)

Reply:
Done